pkg/auth/service.go

```diff
@@ -85,6 +85,7 @@ type ExternalPrincipalsService interface {
 	DeleteUserExternalPrincipal(ctx context.Context, userID, principalID string) error
 	GetExternalPrincipal(ctx context.Context, principalID string) (*model.ExternalPrincipal, error)
 	ListUserExternalPrincipals(ctx context.Context, userID string, params *model.PaginationParams) ([]*model.ExternalPrincipal, *model.Paginator, error)
+	ExternalLogin(ctx context.Context, externalLoginInfo map[string]interface{}) (string, error)
```

Comment:
It needs to be a different authentication service. 
The service doing the login is not RBAC / GIAM, it doesn't have reference to KV. 
That's the part I wanted you to sync with @guy-har.